openedx/core/djangoapps/schedules/resolvers.py

```diff
@@ -142,6 +144,11 @@ def get_schedules_with_target_date_by_bin_and_orgs(
             enrollment__user__in=users,
             enrollment__is_active=True,
             **schedule_day_equals_target_day_filter
+        ).annotate(
+            external_updates_enabled=Exists(query_external_updates(OuterRef('enrollment__user_id'),
+                                                                   OuterRef('enrollment__course_id'))),
+        ).exclude(
+            external_updates_enabled=True,
```

Comment:
This is some fun django magic!

https://docs.djangoproject.com/en/2.2/ref/models/expressions/#filtering-on-a-subquery-expression